pennylane/qaoa/layers.py

```diff
@@ -94,7 +96,7 @@ def circuit(gamma):
 
     """
     # NOTE: op is defined explicitely as validation inside ApproxTimeEvolution needs to be called before checking Hamiltonian
```

Comment:
I'm not so sure what this note really is saying. Is it arguing why first `ApproxTimeEvolution` is initialized and then the `_diagonal_terms` check is being called?
```suggestion
    # NOTE: op is defined explicitly as validation inside ApproxTimeEvolution needs to be called before checking Hamiltonian
```